tests/pubsub_client.rs

```diff
@@ -760,14 +760,36 @@ mod pubsub_client_tests {
             .unwrap();
 
         // Create a subscription to query. Must be created before messages are published.
-        create_dummy_subscription(
+        let subscription = create_dummy_subscription(
             &mut subscription_client,
             emulator.project(),
             subscription_name,
             topic_name,
         )
         .await
-        .unwrap();
+        .unwrap()
+        .into_inner();
+
+        // update the subscription with a DeadLetterPolicy so that the delivery counter engages
```

Comment:
I am not sure I grok this comment.

Reply:
> Does this mean that the delivery counter doesn't work unless you have a dead letter policy?

Exactly; from the [ReceivedMessage](https://docs.rs/ya-gcp/latest/ya_gcp/pubsub/api/struct.ReceivedMessage.html#) docs: `If a DeadLetterPolicy is not set on the subscription, this will be 0`

> does this essentially retry to deliver 5 times then forward to the DLQ?

Yeah i think that's right

Reply:
got it. I think the comment could be slightly more explicit or clear - but, will leave it for you to decide if its worth changing